ArmaforcesMissionBot/Features/Signups/Importer/SignupImporter.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ArmaforcesMissionBot.Exceptions;
+using ArmaforcesMissionBot.Helpers;
+using ArmaforcesMissionBot.Modules;
+using Discord.Commands;
+
+namespace ArmaforcesMissionBot.Features.Signups.Importer {
+    public class SignupImporter {
+        private ICommandContext Context { get; }
+        private CommandService CommandService { get; }
+        private IServiceProvider ServiceProvider { get; }
+        private IModule Module { get; }
+
+        public SignupImporter
+        (
+            ICommandContext context,
+            CommandService commandService,
+            IServiceProvider serviceProvider,
+            IModule module
+        ) {
+            Context = context;
+            CommandService = commandService;
+            ServiceProvider = serviceProvider;
+            Module = module;
+        }
+
+        public async Task ProcessMessage(string message) {
+            var lineBreak = GetLineEnding(message);
+            var lines = ReadLines(message, lineBreak);
+            var commands = ParseCommands(lines, lineBreak);
+
+            if (!commands.Any())
+                await Module.ReplyWithException<InvalidCommandParametersException>("Nie udało się odczytać komend.");
+
+            foreach (var command in commands) await ProcessCommand(command);
```

Comment:
Wouldn't something like this 
```cs
var tasks = commands.Select(c => ProcessCommand(c);
await Tasks.WhenAll(tasks);
```
perform better? 
Ref <https://stackoverflow.com/a/18668733/7411472>

Reply:
Commands must be executed in correct order (at least the first one which starts the signups creation procedure, not sure about the rest). I'll take a look into signups creation 🍝.